onchain/src/tests/nft_quest.cairo

```diff
@@ -0,0 +1,74 @@
+use art_peace::{IArtPeaceDispatcher, IArtPeaceDispatcherTrait};
+use art_peace::quests::nft_quest::NFTMintQuest::NFTMintQuestInitParams;
+use art_peace::nfts::interfaces::{ICanvasNFTStore, NFTMetadata};
+use art_peace::nfts::interfaces::{ICanvasNFTStoreDispatcher, ICanvasNFTStoreDispatcherTrait};
+use art_peace::quests::interfaces::{IQuestDispatcher, IQuestDispatcherTrait};
+use art_peace::tests::art_peace::deploy_with_quests_contract;
+use art_peace::tests::utils;
+use starknet::{ContractAddress, get_caller_address, get_contract_address, contract_address_const};
+use snforge_std as snf;
+use snforge_std::{declare, CheatTarget, ContractClassTrait};
+
+const reward_amt: u32 = 18;
+
+fn SINGLE_CALLDATA() -> Span<felt252> {
+    array![1].span()
+}
+
+fn deploy_nft_contract() -> ContractAddress {
+    let contract = snf::declare("CanvasNFT");
+    let mut calldata = array![];
+    let name: ByteArray = "CanvasNFTs";
+    let symbol: ByteArray = "A/P";
+    name.serialize(ref calldata);
+    symbol.serialize(ref calldata);
+    contract.deploy_at(@calldata, utils::NFT_CONTRACT()).unwrap()
+}
+
+
+fn deploy_nft_quest() -> ContractAddress {
+    let contract = declare("NFTMintQuest");
+
+    let mut nft_quest_calldata = array![];
+    NFTMintQuestInitParams { CanvasNFT: utils::NFT_QUEST_CONTRACT(), reward: reward_amt, }
+        .serialize(ref nft_quest_calldata);
+
+    contract.deploy(@nft_quest_calldata).unwrap()
+}
+
+
+#[test]
+fn test_get_reward() {
+    let contract_address = deploy_nft_quest();
+    let dispatcher = IQuestDispatcher { contract_address };
+    let current_reward = dispatcher.get_reward();
+
+    let test_reward = 18;
+
+    assert(current_reward == test_reward, 'Reward Not set');
+}
+
+#[test]
+fn test_is_claimable() {
+    let contract_address = deploy_nft_quest();
+    let dispatcher = IQuestDispatcher { contract_address };
+    deploy_nft_contract();
+    // let calldata: Array<felt252> = array![0];
+    let test_is_claim = dispatcher.is_claimable(contract_address_const::<1>(), SINGLE_CALLDATA());
+
+    let is_claim = false;
+
+    assert(is_claim == test_is_claim, 'Cannot Claim Mint NFT Quest');
+}
+
+#[test]
+fn test_claim() {
+    let contract_address = deploy_nft_quest();
+    let dispatcher = IQuestDispatcher { contract_address };
```

Comment:
The quest dispatcher should be the art_peace contract. Please use the tests here as reference to fix the issues in this file. Most things should be the same : https://github.com/keep-starknet-strange/art-peace/blob/main/onchain/src/tests/template_quest.cairo

Reply:
Most thing should be the same, but we are switching `deploy_template_quest` with   `deploy_nft_quest` ?

> The quest dispatcher should be the art_peace contract. Please use the tests here as reference to fix the issues in this file. Most things should be the same : https://github.com/keep-starknet-strange/art-peace/blob/main/onchain/src/tests/template_quest.cairo